### PR TITLE
Markdown tests for keeping track of <summary>-related issues.

### PR DESCRIPTION
**Pull request by isoos**
#6276

### CODE REVIEW

File: app/test/shared/markdown_test.dart [hunk 1]
```diff
@@ -255,6 +255,56 @@ void main() {
...
+            'Consider [migrating](https://dart.dev/null-safety/migration-guide).\n'
+            '</details>',
+        '<details>\n'
+            '<summary>\n'
+            'Package language version (indicated by the sdk constraint `>=2.0.0-dev.48.0 <3.0.0`) is less than 2.12.\n'
+            '</summary>\n\n'
+            'Consider [migrating](https://dart.dev/null-safety/migration-guide).\n'
+            '</details>',
+      );
+    });
```
Comment (jonasfj) [inline app/test/shared/markdown_test.dart:298]: ??? aren't you just testing that two strings are equal?

File: app/test/shared/markdown_test.dart [hunk 2]
```diff
@@ -255,6 +255,56 @@ void main() {
...
+  group('<summary>', () {
+    test('FIXME: <summary> must not exists in a list item', () {
+      expect(
+        markdownToHtml('- a<summary>b\n- c'),
+        '<ul>\n'
+        '<li>a<summary>b</summary></li>\n'
+        '<li>c</li>\n'
+        '</ul>\n',
+      );
+    });
```
Comment (jonasfj) [inline app/test/shared/markdown_test.dart:267]:
```suggestion
  testMarkdown(String name, String md, html) {
    test(name, () {
      // TODO: Normalize HTML before comparing
      expect(markdownToHtml(md), html));
    });
  }

    testMarkdown('FIXME: <summary> must not exists in a list item', 
       '- a<summary>b\n- c',
        '<ul>\n'
        '<li>a<summary>b</summary></li>\n'
        '<li>c</li>\n'
        '</ul>\n',
    );
```

Or even better maybe, improve our golden test framework and make it more generically reusable.

**Review by jonasfj — APPROVED** [review]
This is fine as is though.

File: app/test/shared/markdown_test.dart (hunk 1 above)
Comment (isoos) [reply app/test/shared/markdown_test.dart:298]: Yeah, but it shouldn't be, hence the FIXME part in the test's name.

**Pull request merged**